database/ffldb/blockio.go

```diff
@@ -713,36 +720,61 @@ func (s *blockStore) handleRollback(oldBlockFileNum, oldBlockOffset uint32) {
 }
 
 // scanBlockFiles searches the database directory for all flat block files to
-// find the end of the most recent file.  This position is considered the
-// current write cursor which is also stored in the metadata.  Thus, it is used
-// to detect unexpected shutdowns in the middle of writes so the block files
-// can be reconciled.
-func scanBlockFiles(dbPath string) (int, uint32) {
-	lastFile := -1
-	fileLen := uint32(0)
-	for i := 0; ; i++ {
-		filePath := blockFilePath(dbPath, uint32(i))
+// find the first file, last file, and the end of the most recent file.  The
+// position at the last file is considered the current write cursor which is
+// also stored in the metadata.  Thus, it is used to detect unexpected shutdowns
+// in the middle of writes so the block files can be reconciled.
+func scanBlockFiles(dbPath string) (
+	firstFile int, lastFile int, lastFileLen uint32, err error) {
+
+	files, err := filepath.Glob(filepath.Join(dbPath, "*"+blockFileExtension))
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	sort.Strings(files)
+
+	lastFile, firstFile = -1, -1
+	for _, file := range files {
+		// Extract the file number.
+		fileNum := file
+		fileNum = filepath.Base(fileNum)
+		fileNum = strings.TrimSuffix(fileNum, blockFileExtension)
```

Comment:
Inline this.

```suggestion
		fileNum := strings.TrimSuffix(filepath.Base(file), blockFileExtension)
```

If you don't want to inline it I strongly suggest not mutating `fileNum` like this and just create different names for the fragments. If you're worried about overallocation consider allocating stack space outside the loop and only setting each of these values once inside the loop.